sdk/azcore/response.go

```diff
@@ -26,15 +26,16 @@ type Response struct {
 	*http.Response
 }
 
-func (r *Response) payload() []byte {
-	if r.Body == nil {
-		return nil
-	}
+func (r *Response) payload() ([]byte, error) {
 	// r.Body won't be a nopClosingBytesReader if downloading was skipped
 	if buf, ok := r.Body.(*nopClosingBytesReader); ok {
-		return buf.Bytes()
+		return buf.Bytes(), nil
 	}
-	return nil
+	bytesBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+	return bytesBody, nil
```

Comment:
I'm thinking that before we return, we should wrap the result in a `nopClosingBytesReader`.  Either way we should restore the `r.Body` so that customers can still read from it.

Reply:
It also ties in with `removeBOM()`

Reply:
In your first comment, did you mean that we should change the return type for payload()?

Reply:
Nope, you got it right.